public/apps/login/login-page.tsx

```diff
@@ -49,16 +50,34 @@ interface LoginButtonConfig {
   buttonstyle: string;
 }
 
-function redirect(serverBasePath: string) {
-  // navigate to nextUrl
+export function getNextPath(serverBasePath: string) {
   const urlParams = new URLSearchParams(window.location.search);
   let nextUrl = urlParams.get('nextUrl');
   if (!nextUrl || nextUrl.toLowerCase().includes('//')) {
     // Appending the next url with trailing slash. We do so because in case the serverBasePath is empty, we can simply
     // redirect to '/'.
     nextUrl = serverBasePath + '/';
   }
-  window.location.href = nextUrl + window.location.hash;
+  const savedTenant = getSavedTenant();
+  const url = new URL(
+    window.location.protocol + '//' + window.location.host + nextUrl + window.location.hash
+  );
```

Comment:
I'm wondering if there is a better way to do this other than using window.blah - is there some built in way from OSD side?

Reply:
Not that I know of. This maintains the existing behavior of extracting the nextUrl and appending the hash, but it also adds the `security_tenant` url param if it doesn't already exist as part of nextUrl.